recipe/2025-08-recipe-packs.md

```diff
@@ -0,0 +1,379 @@
+            }
+            Radius.Storage/volumes: {
+                recipeKind: 'terraform'
+                recipeLocation: 'https://github.com/project-radius/resource-types-contrib.git//recipes/storage/volumes/kubernetes?ref=v0.48'
+            }
+        }
+    }
+}
+```
+
+The schema for the type would look like:
+
+```yaml
+namespace: Radius.Core
+  types:
+    recipePacks:
+      description: Recipe Pack for grouping and managing related recipes
+      apiVersions:
+        '2026-01-01-preview':
+          schema:
+            type: object
+            properties:
+              name:
+                type: string
+                description: The name of the recipe pack
+              description:
+                type: string
+                description: Description of what this recipe pack provides
+              recipes:
+                type: object
+                description: Map of resource types to their recipe configurations
+                additionalProperties:
+                  type: object
+                  properties:
+                    recipeKind:
+                      type: string
+                      description: The type of recipe (e.g., terraform, bicep)
+                      enum:
+                        - terraform
+                        - bicep
+                    recipeLocation:
+                      type: string
+                      description: URL or path to the recipe source
+                    parameters:
+                      type: object
+                      description: Parameters to pass to the recipe
+                      additionalProperties:
+                        type: any
+                  required:
+                    - recipeKind
+                    - recipeLocation
+            required:
+              - name
+              - recipes
+```
+
+Today the RRT schema does not allow `any` as a type for security reasons. We would have to remove that constraint in bicep tooling and radius so that we can have the recipe parameters as defined above.
+
+// Question: The recipe collection should be curly braces, is that OK (feature spec has []) ? OR we keep it array like below. Map could make it easier to look up using resource type.
+
+"recipes": [
+    {
+      "resourceType": "Radius.Compute/containers",
+      "recipeKind": "terraform",
+      "recipeLocation": "https://example.com/recipes/containers.zip"
+    },
+    {
+      "resourceType": "Radius.Storage/volumes",
+      "recipeKind": "terraform",
+      "recipeLocation": "https://example.com/recipes/volumes.zip"
+    }
+  ]
+
+### API design (if applicable)
+
+We should support CRUDL operations on recipe-pack resource. This should be fairly automatic since recipe packs would be registered as an RRT. Once we register the schema for `Radius.Core/recipePacks` resource type using `rad resource-type create`,  Dynamic RP should be able to dynamically look up the schema and perfom these operations. 
```

Comment:
Question to ponder: should Recipe Packs resource not be modeled as an RRT? Since it's a Radius construct and we don't expect users to need to customize their Recipe Packs?